sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/DoNotUseLiteralBoolInAssertions.cs

```diff
@@ -0,0 +1,120 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.CSharp
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    [Rule(DiagnosticId)]
+    public sealed class DoNotUseLiteralBoolInAssertions : SonarDiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "S2701";
+        private const string MessageFormat = "Remove or correct this assertion.";
+
+        private static readonly DiagnosticDescriptor rule =
+            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
+
+        private static Dictionary<KnownType, HashSet<string>> trackedTypeAndMethods =
+            new Dictionary<KnownType, HashSet<string>>
+            {
+                [KnownType.Xunit_Assert] = new HashSet<string>
+                {
+                    "Equal", "False", "NotEqual", "Same", "StrictEqual", "NotSame"
+                },
+
+                [KnownType.Microsoft_VisualStudio_TestTools_UnitTesting_Assert] = new HashSet<string>
+                {
+                    "AreEqual", "AreNotEqual", "AreSame", "IsFalse", "IsTrue"
+                },
+
+                [KnownType.NUnit_Framework_Assert] = new HashSet<string>
+                {
+                    "AreEqual", "AreNotEqual", "AreNotSame", "AreSame", "False",
+                    "IsFalse", "IsTrue", "That", "True"
+                },
+
+                [KnownType.System_Diagnostics_Debug] = new HashSet<string>
+                {
+                    "Assert"
+                }
+            };
+
+        private static List<SyntaxKind> boolLiterals =
+            new List<SyntaxKind>
+            {
+                SyntaxKind.TrueLiteralExpression,
+                SyntaxKind.FalseLiteralExpression
+            };
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    var invocation = (InvocationExpressionSyntax)c.Node;
+
+                    var symbolInfo = c.SemanticModel.GetSymbolInfo(invocation);
```

Comment:
Will this work with partial classes? Shouldn't we use `that method` @Evangelink wrote?